src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,484 @@
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
+  OnEventCommandExtectations(msg_mobile_response,
+                             mobile_info,
+                             mobile_response,
+                             success,
+                             msg_hmi_response,
+                             command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       OnEvent_HMIResponseUNSUPPORTED_RESOURCE_NOT_SUCCESS) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_hmi_response = CreateFullParamsSO();
+  (*msg_hmi_response)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  MediaAppSetup();
+  DefineHMILevelUIAvailable();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(true));
+  ON_CALL(app_mngr_, ManageHMICommand(_)).WillByDefault(Return(true));
+
+  command->Run();
+
+  DefineInterfaceNotAvailable(am::HmiInterfaces::HMI_INTERFACE_UI);
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
+  OnEventCommandExtectations(msg_mobile_response,
+                             mobile_info,
+                             mobile_response,
+                             success,
+                             msg_hmi_response,
+                             command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       OnEvent_HMIResponseUSER_DISALLOWED_USER_DISALLOWED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_hmi_response = CreateFullParamsSO();
+  (*msg_hmi_response)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::USER_DISALLOWED;
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::USER_DISALLOWED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  MediaAppSetup();
+  DefineHMILevelUIAvailable();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
+
+  command->Run();
+
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
+  OnEventCommandExtectations(msg_mobile_response,
+                             mobile_info,
+                             mobile_response,
+                             success,
+                             msg_hmi_response,
+                             command);
+}
+
```

Comment:
@vlantonov 
Additional test case is added.
Done in 1456832e723cd462c70ff1e55e4818cce6867ef7